c/src/ml-api-inference-single.c

```diff
@@ -830,6 +830,36 @@ ml_single_open_custom (ml_single_h * single, ml_single_preset * info)
     goto error;
   }
 
+  if (nnfw == ML_NNFW_TYPE_NNTR_INF) {
+    if (!in_tensors_info || !out_tensors_info) {
+      if (!in_tensors_info) {
+        ml_tensors_info_h in_info;
+        status = ml_tensors_info_create (&in_info);
+        if (status != ML_ERROR_NONE) {
+          goto error;
+        }
+
+        status = ml_tensors_info_set_count (in_info, 0);
+        if (status != ML_ERROR_NONE) {
+          ml_tensors_info_destroy (in_info);
+          goto error;
+        }
+
+        /// @note ml_single_set_input_info is not feasible as it checks on the num_tensors
+        status = ml_single_set_gst_info (single_h, in_info);
+        ml_tensors_info_destroy (in_info);
+        if (status != ML_ERROR_NONE) {
+          goto error;
+        }
+      } else {
+        status = ml_single_set_input_info (single_h, info->input_info);
```

Comment:
What about using `in_tensors_info` instead of `info->input_info`?

Reply:
updated!